### PR TITLE
Added a new menu item and a new dialog(circular)

### LINKED ISSUE #5474: New menu item and new dialogue to define circular
I propose a new menu item in the Prepare menu.  It is called **_Column: Define_**.   and is below Column: Date and above Column: Reshape.
It can start with just two items.  The first is a copy (at least for now) of Prepare > Data Frame > Convert Columns and is also called **_Convert Columns_**.  Then there is a line.  Under the line is Circular.
This is a new dialogue.  There is the usual data selector restricted to numeric columns.  There is a multiple receiver with the label **To Circular**.  (I can't think of many situations when you will want more than one variable.  So if it is more work, then make it a single receiver.)

There are two sets of the ordinary radio buttons.
One is called Units.  It has 3 options, namely Radians, Degrees, Hours with Radians the default.
The second is Modulo.  It has 3 options, namely Asis, pi, 2pi with 2pi the default.

There are other arguments, but they are mainly for the plotting routines in the package and w are not using those.  So stop there, at least for now.

This just issues the command circular::circular(x, modulo="2pi", units="radians")

Our default is 2pi, so we have to give that, because the package default is asis.  Our default of radians is the same as the package.

If it is a multiple receiver then the other controls remain simple, i.e. if one variable is in degrees and the other is in radians, then you use the dialogue twice.

### PR DESCRIPTION
**Pull request by Wycklife**
Fixes #5474

@africanmathsinitiative/developers This is ready for review

### CODE REVIEW

**Comment by Wycklife** [comment]
@shadrackkibet Tab position has been set too

**Comment by shadrackkibet** [comment]
Frm main changes should be done in a separate branch!

**Comment by Wycklife** [comment]
I just enabled the dialog, I thought it's a small change

**Comment by shadrackkibet** [comment]
@dannyparsons this is working fine. This is ready for you to review.

**Comment by shadrackkibet** [comment]
@rdstern this needs some testing. Have a look at it.

**Comment by maxwellfundi** [comment]
Thank you @shadrackkibet for the review. 
@Wycklife any form main changes should be in new branch, done separately. @africanmathsinitiative/developers  we should keep this in mind.

**Comment by Wycklife** [comment]
@maxwellfundi Thank you for the observation. Is there any change to be made?

File: instat/dlgCircular.vb [hunk 1]
```diff
@@ -1,3 +1,94 @@
...
+        ucrPnlUnits.AddRadioButton(rdoRadians, Chr(34) & "radians" & Chr(34))
+
+        ucrSaveCircularColumn.SetPrefix("Circ")
+        ucrSaveCircularColumn.SetSaveTypeAsColumn()
+        ucrSaveCircularColumn.SetDataFrameSelector(ucrSelectorCircular.ucrAvailableDataFrames)
+        ucrSaveCircularColumn.SetLabelText("New Column Name:")
+        ucrSaveCircularColumn.SetIsComboBox()
+    End Sub
+
+    Private Sub setDefaults()
```
Comment (dannyparsons) [inline instat/dlgCircular.vb]: Capital `S` at the start

File: instat/dlgCircular.vb [hunk 2]
```diff
@@ -1,3 +1,94 @@
...
+        clsCircularFunction.SetRCommand("circular")
+
+        clsCircularFunction.AddParameter("units", Chr(34) & "radians" & Chr(34), iPosition:=2)
+        clsCircularFunction.AddParameter("modulo", Chr(34) & "2pi" & Chr(34), iPosition:=1)
+
+        clsCircularFunction.SetAssignTo(ucrSaveCircularColumn.GetText, strTempDataframe:=ucrSelectorCircular.ucrAvailableDataFrames.cboAvailableDataFrames.Text, strTempColumn:=ucrSaveCircularColumn.GetText)
+        ucrBase.clsRsyntax.SetBaseRFunction(clsCircularFunction)
+    End Sub
+
+    Private Sub setRcodeforControls(breset As Boolean)
```
Comment (dannyparsons) [inline instat/dlgCircular.vb]: Capital `S` at the start.

File: instat/dlgCircular.vb [hunk 3]
```diff
@@ -1,3 +1,94 @@
...
+        ucrPnlModulo.AddRadioButton(rdoAsis, Chr(34) & "asis" & Chr(34))
+        ucrPnlModulo.AddRadioButton(rdo2Pi, Chr(34) & "2pi" & Chr(34))
+        ucrPnlModulo.AddRadioButton(rdoPi, Chr(34) & "pi" & Chr(34))
+
+        ucrPnlUnits.SetParameter(New RParameter("units", 2))
+        ucrPnlUnits.AddRadioButton(rdoDegrees, Chr(34) & "degrees" & Chr(34))
+        ucrPnlUnits.AddRadioButton(rdoHours, Chr(34) & "hours" & Chr(34))
+        ucrPnlUnits.AddRadioButton(rdoRadians, Chr(34) & "radians" & Chr(34))
+
+        ucrSaveCircularColumn.SetPrefix("Circ")
```
Comment (dannyparsons) [inline instat/dlgCircular.vb]: Lowercase `c` for column names (following the [tidyverse style guide](https://style.tidyverse.org/syntax.html)).

**Comment by Wycklife** [comment]
@dannyparsons Changes implemented

**Comment by dannyparsons** [comment]
`SetDefaults` and `SetRcodeforControls` seem to still be in lowercase in the places they are called. Is this causing errors? Even if not better to update them too.

**Comment by Wycklife** [comment]
@dannyparsons Sorry for that, it's now okay.

**Pull request merged**